scripts/validate-requirements.sh

```diff
@@ -0,0 +1,274 @@
+        echo "Visit https://git-scm.com/book/en/v2/Getting-Started-Installing-Git and follow the instructions to install Git."
+        ERRORS+=$'git not found.\n'
+    else
+        GIT_CURRENT_VERSION=$(git version | awk '{print $3}')
+        if [ "$(compare_version "$GIT_CURRENT_VERSION" "$GIT_VERSION")" -eq 2 ]; then
+            echo "An incompatibly was found in git version."
+            echo "Version required is at least $GIT_VERSION"
+            echo "Visit https://git-scm.com/book/en/v2/Getting-Started-Installing-Git and follow the instructions to install Git."
+            ERRORS+=$'git version is incompatible.\n'
+        fi
+    fi
+
+    if ! git config init.defaultBranch | grep "main" >/dev/null ; then
+        echo "git defaul branch must be configured."
+        echo "See the instructions at https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/docs/TROUBLESHOOTING.md#default-branch-setting ."
+        ERRORS+=$'git default branch must be configured.\n'
+    fi
+}
+
+# Validate the Configuration of the Gcloud CLI
+function validate_gcloud_configuration(){
+
+    END_USER_CREDENTIAL_OUTPUT="$(gcloud config get-value account 2>&1 >/dev/null)"
+    if [ "$(echo "$END_USER_CREDENTIAL_OUTPUT" | grep -c unset)" -eq 1 ]; then
+        echo "You must configure an End User Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud end user credential not configured.\n'
+    fi
+
+    APPLICATION_DEFAULT_CREDENTIAL_OUTPUT="$(gcloud auth application-default print-access-token 2>&1 >/dev/null)"
+    if [ "$(echo "$APPLICATION_DEFAULT_CREDENTIAL_OUTPUT" | grep -c 'Could not automatically determine credentials')" -eq 1 ]; then
+        echo "You must configure an Application Default Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud application default credential not configured.\n'
+    fi
+}
+
+# Function to validate the Configuration of the Gcloud CLI
+function validate_credential_roles(){
+    check_org_level_roles "$END_USER_CREDENTIAL" "$ORGANIZATION_ID"
+    check_billing_account_roles "$END_USER_CREDENTIAL" "$BILLING_ACCOUNT"
+}
+
+# Verifies wheter a user is assigned to the expected Orgzanization Level roles
+function check_org_level_roles(){
+
+    ORG_LEVEL_ROLES_OUTPUT=$(
+        gcloud organizations get-iam-policy "$2" \
+        --filter="bindings.members:$1" \
+        --flatten="bindings[].members" \
+        --format="table(bindings.role)" 2>/dev/null)
+
+    lines=$(echo "$ORG_LEVEL_ROLES_OUTPUT" | grep -o -e roles/resourcemanager.folderCreator -e roles/resourcemanager.organizationAdmin -e roles/orgpolicy.policyAdmin | wc -l)
```

Comment:
Can we move expected roles to a bash array so we can keep them separate from the function?
Also would using jq be simpler here?

Reply:
@bharathkkb I moved the roles to a specific constant variable, and now the checking if more dynamic. 
I couldn't visualize how JQ could help me better at this process. I think the new way I did is even simplier.